### PR TITLE
[fix](merge-cloud) fix inconsistency between cloud mode and local mode for 2PC

### PR DESCRIPTION
**Pull request by liaoxin01**
## Proposed changes

Issue Number: close #xxx

1. fix test_stream_load case
2. When the labels are the same, return the associated txn id
3. Repeat commit and return error message for 2pc

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by liaoxin01** [comment]
run buildall

File: cloud/src/meta-service/meta_service_txn.cpp [hunk 1]
```diff
@@ -1969,4 +1969,119 @@ void MetaServiceImpl::clean_txn_label(::google::protobuf::RpcController* control
...
+    if (!label_pb.ParseFromArray(label_val.data(), label_val.size() - VERSION_STAMP_LEN)) {
+        code = MetaServiceCode::PROTOBUF_PARSE_ERR;
+        ss << "label_pb->ParseFromString() failed, label=" << label;
+        msg = ss.str();
+        return;
+    }
+    if (label_pb.txn_ids_size() == 0) {
+        code = MetaServiceCode::TXN_ID_NOT_FOUND;
+        ss << "transaction not found, label=" << label;
+        return;
```
Comment (gavinchou) [inline cloud/src/meta-service/meta_service_txn.cpp:2035]: missing msg update?

File: cloud/src/meta-service/meta_service_txn.cpp [hunk 2]
```diff
@@ -1969,4 +1969,119 @@ void MetaServiceImpl::clean_txn_label(::google::protobuf::RpcController* control
...
+        for (auto status : request->status()) {
+            if (cur_txn_info.status() == status) {
+                response->set_txn_id(cur_txn_id);
+                return;
+            }
+        }
+    }
+    code = MetaServiceCode::TXN_ID_NOT_FOUND;
+    ss << "transaction not found, label=" << label;
+    return;
```
Comment (gavinchou) [inline cloud/src/meta-service/meta_service_txn.cpp:2084]: missing msg update?

**Review by gavinchou — CHANGES_REQUESTED** [review]

File: gensrc/proto/cloud.proto [hunk 3]
```diff
@@ -687,6 +687,18 @@ message GetTxnResponse {
...
+message GetTxnIdRequest {
+    optional string cloud_unique_id = 1; // For auth
+    optional int64 db_id = 2;
+    optional string label = 3;
+    repeated TxnStatusPB status = 4;
+}
+
+message GetTxnIdResponse {
+    optional MetaServiceResponseStatus status = 1;
+    optional int64 txn_id = 2;
```
Comment (gavinchou) [inline gensrc/proto/cloud.proto:699]: repeated?

File: cloud/src/meta-service/meta_service_txn.cpp [hunk 4]
```diff
@@ -1969,4 +1969,119 @@ void MetaServiceImpl::clean_txn_label(::google::protobuf::RpcController* control
     code = MetaServiceCode::OK;
 }
 
+void MetaServiceImpl::get_txn_id(::google::protobuf::RpcController* controller,
```
Comment (gavinchou) [inline cloud/src/meta-service/meta_service_txn.cpp:1972]: Add sufficient comment to describe what does it do.

File: cloud/src/meta-service/meta_service_txn.cpp
Comment (gavinchou) [inline cloud/src/meta-service/meta_service_txn.cpp:1]: pls add UT to cover all the newly added code: line/branch coverage 90+%

**Review by gavinchou — CHANGES_REQUESTED** [review]

**Comment by liaoxin01** [comment]
run buildall

File: gensrc/proto/cloud.proto (hunk 3 above)
Comment (liaoxin01) [reply gensrc/proto/cloud.proto:699]: only return one txn_id.

**Review by gavinchou — APPROVED** [review]

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**